libraries/plugins/query_txid/query_txid_plugin.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2019 GXChain and zhaoxiangfei、bijianing97 .
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <graphene/query_txid/query_txid_plugin.hpp>
+#include <fc/io/fstream.hpp>
+#include <graphene/chain/transaction_entry_object.hpp>
+#include <queue>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <leveldb/db.h>
+#include <leveldb/write_batch.h>
+#include <fc/signals.hpp>
+
+namespace graphene{ namespace query_txid{
+
+namespace detail
+{
+using namespace leveldb;
+class query_txid_plugin_impl
+{
+   public:
+      query_txid_plugin_impl(query_txid_plugin &_plugin)
+         : _self(_plugin)
+      {
+      }
+      ~query_txid_plugin_impl()
+      {
+      }
+
+      void collect_txid_index(const signed_block &b);
+
+      graphene::chain::database &database()
+      {
+       return _self.database();
+      }
+      void init();
+      static optional<trx_entry_object> query_trx_by_id(std::string txid);
+      std::string db_path = "trx_entry.db";
+      uint64_t limit_batch = 1000; 
+
+   private:
+      query_txid_plugin &_self;
+
+      fc::signal<void()> sig_db_write;
```

Comment:
The use of signals here looks unnecessarily complicated to me. You only connect one consumer to each signal, and trigger each signal only in one place. Why not call the consumer there directly?